x-pack/plugins/observability_solution/observability_ai_assistant/server/rule_connector/index.ts

```diff
@@ -0,0 +1,248 @@
+  initResources: (request: KibanaRequest) => Promise<ObservabilityAIAssistantRouteHandlerResources>
+): Promise<ConnectorTypeExecutorResult<unknown>> {
+  const state = execOptions.params.alertState;
+  if (state !== 'new') {
+    // system connector action frequency can't be user configured. we use this
+    // flag as dedup mechanism to prevent triggering the same worfklow again.
+    // while recovered state is a valid trigger we only allow new alerts for now
+    return { actionId: execOptions.actionId, status: 'ok' };
+  }
+
+  const request = execOptions.request!;
+  const resources = await initResources(request);
+  const client = await resources.service.getClient({ request });
+  const functionClient = await resources.service.getFunctionClient({
+    signal: new AbortController().signal,
+    resources,
+    client,
+    screenContexts: [],
+  });
+  const actionsClient = await (
+    await resources.plugins.actions.start()
+  ).getActionsClientWithRequest(request);
+
+  const systemMessage = buildSystemMessage(functionClient, execOptions.params.rule, state);
+
+  await lastValueFrom(
+    client
+      .complete({
+        functionClient,
+        persist: true,
+        isPublic: true,
+        connectorId: execOptions.params.connector,
+        signal: new AbortController().signal,
+        kibanaPublicUrl: (await resources.context.core).coreStart.http.basePath.publicBaseUrl,
+        messages: [
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.System,
+              content: systemMessage,
+            },
+          },
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.User,
+              content: execOptions.params.message,
+            },
+          },
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.Assistant,
+              content: '',
+              function_call: {
+                name: 'get_connectors',
+                arguments: JSON.stringify({}),
+                trigger: MessageRole.Assistant as const,
+              },
+            },
+          },
+          {
+            '@timestamp': new Date().toISOString(),
+            message: {
+              role: MessageRole.User,
+              name: 'get_connectors',
+              content: JSON.stringify({
+                connectors: await actionsClient.getAll().then((connectors) => {
+                  return connectors.map((connector) => {
+                    if (connector.actionTypeId === '.slack') {
+                      return {
+                        ...connector,
+                        // TODO: figure out how to parse ParamsSchema at
+                        // x-pack/plugins/stack_connectors/server/connector_types/slack/index.ts
+                        params: {
+                          message: 'string',
+                        },
```

Comment:
I couldn't fix ts errors when using [joi-to-json](https://github.com/kenspirit/joi-to-json#readme) to extract usable schema

Reply:
I'm not sure it's possible to go from a connector type to get it's schema validator - currently most are using `config-schema` and just provide a `validate()` function which either returns a string with an error, or nothing when valid.

In theory you could find the validators for each connector type and manually reference them.  For instance, here's the Slack API connector type, which includes the validators.  That's an interesting one, because it points out the complete validation can require validating based on on other configuration (for instance, that the hostname used in the config URL is allowed).

If we're going to be doing more like this, we may want to expose the validators in an easier way ...

https://github.com/elastic/kibana/blob/1c38891dfcdb875d39cbd0e3970043e44689b1df/x-pack/plugins/stack_connectors/server/connector_types/slack_api/index.ts#L35-L54

Reply:
The idea was to get an easy connector as a first iteration, and using webhook is something that could be used way more extensively by users in this connector context. For example one could ask to post a message to a webhook as free text in the ai assistant connector without having to rely on a connector being configured.
We definitely want to cover more connectors and I'll follow up with the team to see how we can get the schema and validator exposed correctly

Reply:
We probably want to open an issue to track this.  Beyond the validators, you'll want all the connector/action UX experience, so for the Slack-api connector, it will let you use channel names instead of the ids.  I'm not really sure where we are at, in terms of being able to use the actions UX stuff outside of alerting rules ...